DSpace/DSpace
# Fix inherit policies at item move

## References
* Fixes #8987
* Requires Angular PR: https://github.com/DSpace/dspace-angular/pull/2423

## Description
In `ItemServiceImpl`, adjusts the behavior of `inheritCollectionDefaultPolicies` such that (if given additional boolean parameter is true), all read policies attached to the given item, its bundles and its bitstreams are removed and replaced by the collection's default policies (default item read policy for item and bundles, default bitstream read policy for bitstreams). This only happens if the collection has these default policies.

## Instructions for Reviewers
### Changes
* `ItemService` and `ItemServiceImpl`: Added variants of the methods `inheritCollectionDefaultPolicies`, `adjustBundleBitstreamPolicies` and `adjustItemPolicies` with an additional boolean parameter that indicates whether read policies have to be overridden.
* `ItemServiceTest`: Added new tests `testMoveItemToCollectionWithMoreRestrictiveItemReadPolicy` and `testMoveItemToCollectionWithMoreRestrictiveBitstreamReadPolicy` that test the new behavior.
* `InstallItemServiceImpl`: Adjusted the call to `itemService.inheritCollectionDefaultPolicies` in `installItem` to use the non-overriding variants of the `ItemService` methods.

### Testing
* Create a regular and restricted collection (restricted collection does not have group "Anonymous" on its default item/bitstream read policy, but rather for example the "Administrators" group).
* Create an item in the regular collection, check its RP, verify it has anonymous read RP
* Move the item to the restricted collection with inheritPolicies=true
* Check the item’s RP again, now the only read RP should be the one for the "Administrators" group inherited from collection. The original RP for "Anonymous" group should be gone.
* Move item back to regular collection with inheritPolicies=true => Verify it has Anon ITEM_READ RP again, no longer has Admin one (so back to publicly available item)

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
